Add Improving FTP Client APIs Proposal (#1398)
ballerina-platform/ballerina-spec: beps/lib-ftp/1397_improving_apis.md

## Goals

1. Introduce new key APIs that most File Integration Solutions offer in the world, however is not yet available directly in the Ballerina FTP Client.

## Non-Goals

1. This proposal does not aim to any changes to existing APIs. It will only add a few new APIs to the current stack.

## Motivation

This aims to improve the FTP Client of Ballerina, hence improving File Integration capabilities of BI, which would create more opportunities for marketing BI for advanced file related integrations

## Design

> The proposed. changes will introduce 3 new APIs which are identified as essential to encompass a minimal File Inetgration experience to the user.

Reviewer: ```suggestion
The proposed changes will introduce 3 new APIs, identified as essential to provide a minimal File Integration experience for the user.
```

Author: Done
